test/snippet/alphabet/composite/semialphabet_any.cpp

```diff
@@ -0,0 +1,34 @@
+//! [example]
+#include <iostream>
+
+#include <seqan3/alphabet/aminoacid/all.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/convert.hpp>
+
+#include <seqan3/alphabet/composite/semialphabet_any.hpp>
+
+using namespace seqan3;
+
+int main()
+{
+    // Original alphabet
+    aa10li_vector v1{"AVRSTXOUB"_aa10li};
+    debug_stream << v1 << "\n"; // AIKAAAKCB
+
+    // Type-erased semi-alphabet
+    auto v2 = v1 | view::convert<semialphabet_any<10> >;
+
+    // Conversion back to original alphabet
+    aa10li_vector v3 = v2;
+    debug_stream << v3 << "\n"; // AIKAAAKCB
+
+    // Conversion to another alphabet
+    // This is possible however should be used with care since there will be no warning given when attempting to convert
+    // a semialphabet_any into an alphabet that is not the original alphabet. This may lead to errors when transforming
+    // the semialphabet_any into an alphabet type that is not comparable to the original type.
+    aa10murphy_vector v4 = v2;
```

Comment:
Hm, but this is what we want people not to do!

I would maybe make the example like this (roughly sketched and not tested, bad grammar)
```cpp
// print is a template and gets instantiated two times,
// because the behaviour is different for both types
template <typename rng_t>
void print(rng_t && r)
{
    seqan3::debug_stream << r << '\n';
}

// algorithm is not a template, only one instance is generated by the compiler;
// type information is encoded via a run-time parameter
void algorithm(std::vector<semialphabet_any<10>> & r, bool is_murphy)
{
    // a silly algorithm example that replaces rank 0 with rank 1:
    for (auto & v : r)
        if (seqan3::to_rank(v) == 0)
            seqan3::assign_rank_to(1, v);

    // here we reify the type for printing
    if (is_murphy)
        print(v | seqan3::view::convert<aa10murphy>);
    else
        print(v | seqan3::view::convert<aa10li>);
}

// two instances of algo_pre exist; they type erase the different arguments to the same type
// and encode the type information as a run-time parameter
void algo_pre(aa10li_vector const & v)
{
    seqan3::semialphabet_any<10> tmp = v | seqan3::view::convert<semialphabet_any<10>>;
    algo(tmp, false);
}
void algo_pre(aa10murphy_vector const & v)
{
    seqan3::semialphabet_any<10> tmp = v | seqan3::view::convert<semialphabet_any<10>>;
    algo(tmp, true);
}

int main()
{
    seqan33::aa10li_vector v1 {...};
    algo_pre(v1);

    seqan3::aa10murphy_vector v2 {...};
    algo_pre(v2);
}
```

And explanation:
```
The example illustrates how we can reduce the usage of templates
(or the amount of different instantiations) via type erasure. Having only
one function generated for `algo()` is the only benefit of  using `semialphabet_any`
here. Of course this only makes sense for your application if the part of the program
that is agnostic of the character representation (your equivalent of `algo()`) is
substantially larger than the specific parts – and if compile-time and/or 
size of the exectuble are a concern.
```

Reply:
BTW: we don't use `using namespace seqan3;` in the snippets anymore...